src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/ExcludeDateService.kt

```diff
@@ -0,0 +1,188 @@
+package uk.gov.justice.digital.hmpps.visitscheduler.service
+
+import jakarta.validation.ValidationException
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.visitscheduler.dto.ExcludeDateDto
+import uk.gov.justice.digital.hmpps.visitscheduler.dto.visitnotification.PrisonDateBlockedDto
+import uk.gov.justice.digital.hmpps.visitscheduler.dto.visitnotification.SessionDateBlockedDto
+import uk.gov.justice.digital.hmpps.visitscheduler.model.entity.IExcludeDate
+import uk.gov.justice.digital.hmpps.visitscheduler.model.entity.Prison
+import uk.gov.justice.digital.hmpps.visitscheduler.model.entity.PrisonExcludeDate
+import uk.gov.justice.digital.hmpps.visitscheduler.model.entity.session.SessionTemplate
+import uk.gov.justice.digital.hmpps.visitscheduler.model.entity.session.SessionTemplateExcludeDate
+import uk.gov.justice.digital.hmpps.visitscheduler.repository.PrisonExcludeDateRepository
+import uk.gov.justice.digital.hmpps.visitscheduler.repository.SessionTemplateExcludeDateRepository
+import java.time.LocalDate
+import java.util.stream.Collectors
+
+@Service
+@Transactional
+class ExcludeDateService(
+  private val messageService: MessageService,
+  private val visitNotificationEventService: VisitNotificationEventService,
+  private val telemetryClientService: TelemetryClientService,
+  private val prisonExcludeDateRepository: PrisonExcludeDateRepository,
+  private val sessionTemplateExcludeDateRepository: SessionTemplateExcludeDateRepository,
+) {
+  companion object {
+    val LOG: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  @Throws(ValidationException::class)
+  fun addExcludeDate(prison: Prison, excludeDateDto: ExcludeDateDto) {
+    addPrisonExcludeDate(prison, excludeDateDto)
+  }
+
+  @Throws(ValidationException::class)
+  fun addExcludeDate(sessionTemplate: SessionTemplate, excludeDateDto: ExcludeDateDto) {
+    addSessionTemplateExcludeDate(sessionTemplate, excludeDateDto)
+  }
+
+  @Throws(ValidationException::class)
+  fun removeExcludeDate(prison: Prison, excludeDateDto: ExcludeDateDto) {
```

Comment:
Maybe use unique naming instead of overloading? As technically they're separate thing (removePrisonExcludeDate and removeSessionExcludeDate)

Reply:
Personally I prefer the overloaded methods - they do in turn call the removePrisonExcludeDate and removeSessionExcludeDate methods.